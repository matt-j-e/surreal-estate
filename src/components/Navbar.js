import React from "react";
import { Link } from "react-router-dom";
import "../styles/Navbar.css";

const Navbar = () => {
  return (
    <header>
      <div className="branding">
        <img
          className="logo"
          src="./surreal-logo.png"
          alt="Surreal Estate logo"
          width="90"
        />
        <h1 className="branding-name">Surreal<span>Estate</span></h1>
      </div>
      <nav className="navbar">
        <ul className="navbar-links">
          <li className="navbar-links-item">
            <Link to="/">View Properties</Link>
          </li>
          <li className="navbar-links-item">
            <Link to="/add-property">Add a Property</Link>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Navbar;
