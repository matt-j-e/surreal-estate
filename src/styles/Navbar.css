/* header {
    background: white;
    border-bottom: #666;
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 90px
} */

/* .branding {
    position: relative;
} */

/* .logo {
    position: absolute;
    top: -9px;
    left: 0;
    outline: 1px solid red;
    height: 60px;
    width: 60px;
} */

/* .branding-name {
    font-weight: 200;
    font-size: 3rem;
    letter-spacing: -0.1rem;
    color: black;
    text-transform: uppercase;
    margin: 0 0 0 60px;
    outline: 1px solid red;
} */

/* .branding-name > span {
    color: #daad65;
    font-weight: 400;
} */

/* .navbar {
    outline: 1px solid red;
    display: flex;
    flex-direction: column;
    justify-content: center;
} */

/* .navbar-links {
    margin: 0;
    padding: 0;
    list-style-type: none;
    outline: 1px solid blue;
} */

/* .navbar-links-item {
    margin: 0;
    padding: 0 0 0 3rem;
    text-transform: uppercase;
    display: inline-block;
} */

/* .navbar-links-item a {
    color: inherit;
    text-decoration: none;
} */

/* .navbar-links-item a:hover {
    color: rgb(218, 173, 101);
} */