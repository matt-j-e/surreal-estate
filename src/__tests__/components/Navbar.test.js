import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import Navbar from "../../components/Navbar";

describe("Navbar", () => {
  it("Renders the navbar correctly", () => {
    const { asFragment } = render(
      <Router>
        <Navbar />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
