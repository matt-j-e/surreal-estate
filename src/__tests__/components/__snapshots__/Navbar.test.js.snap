// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Renders the navbar correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-bdvvaa cqNbLe"
  >
    <div
      class="sc-gsDJrp dmkRxm branding"
    >
      <img
        alt="Surreal Estate logo"
        class="sc-dkPtyc dXKpfu logo"
        src="./surreal-logo.png"
        width="90"
      />
      <h1
        class="sc-hKwCoD iVKNRI"
      >
        Surreal
        <span>
          Estate
        </span>
      </h1>
    </div>
    <nav
      class="sc-eCImvq jbzXgP"
    >
      <ul
        class="sc-jRQAMF dQiAco"
      >
        <li
          class="sc-gKckTs llrDbz"
        >
          <a
            href="/"
          >
            View Properties
          </a>
        </li>
        <li
          class="sc-gKckTs llrDbz"
        >
          <a
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
